File: backend/src/main/java/com/williamdsw/semsys/domain/dto/PersonNewDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import org.hibernate.validator.constraints.Length;

public class PersonNewDTO implements Serializable
{
	// FIELDS
	
	private static final long serialVersionUID = 1L;
	protected Integer id;
	
	@NotEmpty (message = "Name is required")
	@Length (min = 5, max = 120, message = "Name must be between 5 and 120 characters")
	protected String name;
	
	@NotEmpty (message = "Email is required")
	@Email (message = "Invalid email")
	protected String email;
	
	@NotEmpty (message = "SSN is required")
	protected String socialSecurityNumber;
	
	@NotEmpty (message = "Password is required")
	protected String password;
	
	@NotEmpty (message = "At least one phone number is required")
	protected Set<String> phoneNumbers = new HashSet<>();
	
	// -- ADDRESS FIELDS
	
	@NotEmpty (message = "Street is required")
	protected String street;
	
	@NotEmpty (message = "Number is required")
	protected String number;
	protected String complement;
	
	@NotEmpty (message = "Zip Code is required")
	protected String zipCode;
	
	// -- CITY FIELDS
	
	protected Integer cityId;
	
	// CONSTRUCTOR
	
	public PersonNewDTO () {}
	
	// GETTERS / SETTERS

	public Integer getId () 
	{
		return id;
	}
	
	public void setId (Integer id) 
	{
		this.id = id;
	}
	
	public String getName () 
	{
		return name;
	}
	
	public void setName (String name) 
	{
		this.name = name;
	}
	
	public String getEmail () 
	{
		return email;
	}
	
	public void setEmail (String email) 
	{
		this.email = email;
	}
	
	public String getSocialSecurityNumber () 
	{
		return socialSecurityNumber;
	}
	
	public void setSocialSecurityNumber (String socialSecurityNumber) 
	{
		this.socialSecurityNumber = socialSecurityNumber;
	}
	
	public String getPassword () 
	{
		return password;
	}
	
	public void setPassword (String password) 
	{
		this.password = password;
	}
	
	public Set<String> getPhoneNumbers () 
	{
		return phoneNumbers;
	}
	
	public void setPhoneNumbers (Set<String> phoneNumbers) 
	{
		this.phoneNumbers = phoneNumbers;
	}
	
	public String getStreet () 
	{
		return street;
	}
	
	public void setStreet (String street) 
	{
		this.street = street;
	}
	
	public String getNumber () 
	{
		return number;
	}
	
	public void setNumber (String number) 
	{
		this.number = number;
	}
	
	public String getComplement () 
	{
		return complement;
	}
	
	public void setComplement (String complement) 
	{
		this.complement = complement;
	}
	
	public String getZipCode () 
	{
		return zipCode;
	}
	
	public void setZipCode (String zipCode) 
	{
		this.zipCode = zipCode;
	}
	
	public Integer getCityId () 
	{
		return cityId;
	}
	
	public void setCityId (Integer cityId) 
	{
		this.cityId = cityId;
	}
}

File: frontend/src/app/models/domain/new-dto/person.new.dto.ts

export abstract class PersonNewDTO {

    protected id: number;
    protected name: string;
    protected email: string;
    protected socialSecurityNumber: string;
    protected password: string;
    protected phoneNumbers: string[] = [];

    protected street: string;
    protected number: string;
    protected complement: string;
    protected zipCode: string;

    protected cityId: number;

    constructor() {}

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getEmail(): string {
        return this.email;
    }

    public setEmail(email: string) {
        this.email = email;
    }

    public getSocialSecurityNumber(): string {
        return this.socialSecurityNumber;
    }

    public setSocialSecurityNumber(socialSecurityNumber: string) {
        this.socialSecurityNumber = socialSecurityNumber;
    }

    public getPassword(): string {
        return this.password;
    }

    public setPassword(password: string) {
        this.password = password;
    }

    public getPhoneNumbers(): string[] {
        return this.phoneNumbers;
    }

    public setPhoneNumbers(phoneNumbers: string[]) {
        this.phoneNumbers = phoneNumbers;
    }

    public getStreet(): string {
        return this.street;
    }

    public setStreet(street: string) {
        this.street = street;
    }

    public getNumber(): string {
        return this.number;
    }

    public setNumber(value: string) {
        this.number = value;
    }

    public getComplement(): string {
        return this.complement;
    }

    public setComplement(complement: string) {
        this.complement = complement;
    }

    public getZipCode(): string {
        return this.zipCode;
    }

    public setZipCode(zipCode: string) {
        this.zipCode = zipCode;
    }

    public getCityId(): number {
        return this.cityId;
    }

    public setCityId(cityId: number) {
        this.cityId = cityId;
    }
}
